adaptive/src/main/java/com/google/accompanist/adaptive/FoldAwareColumn.kt

```diff
@@ -0,0 +1,383 @@
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.window.layout.DisplayFeature
+import androidx.window.layout.FoldingFeature
+import kotlin.math.roundToInt
+
+/**
+ * A simplified version of [Column] that places children in a fold-aware manner.
+ *
+ * The layout starts placing children from the top of the available space. If there is a horizontal
+ * [separating](https://developer.android.com/reference/kotlin/androidx/window/layout/FoldingFeature#isSeparating())
+ * fold present in the window, then the layout will check to see if any children overlap the fold.
+ * If a child would overlap the fold in its current position, then the layout will increase its
+ * y coordinate so that the child is now placed below the fold, and any subsequent children will
+ * also be placed below the fold.
+ *
+ *
+ * @param displayFeatures a list of display features the device currently has
+ * @param modifier an optional modifier for the layout
+ * @param foldPadding the optional padding to add around a fold
+ * @param horizontalAlignment the horizontal alignment of the layout's children.
+ */
+@Composable
+public fun FoldAwareColumn(
+    displayFeatures: List<DisplayFeature>,
+    modifier: Modifier = Modifier,
+    foldPadding: PaddingValues = PaddingValues(),
+    horizontalAlignment: Alignment.Horizontal = Alignment.Start,
+    content: @Composable FoldAwareColumnScope.() -> Unit,
+) {
+    // Extract folding feature if horizontal and separating
+    val fold = displayFeatures.find {
+        it is FoldingFeature && it.orientation == FoldingFeature.Orientation.HORIZONTAL && it.isSeparating
+    } as FoldingFeature?
+
+    // Calculate fold bounds in pixels (including any added fold padding)
+    val foldBoundsPx = with(LocalDensity.current) {
+        val topPaddingPx = foldPadding.calculateTopPadding().roundToPx()
+        val bottomPaddingPx = foldPadding.calculateBottomPadding().roundToPx()
+
+        fold?.bounds?.let {
+            Rect(
+                left = it.left.toFloat(),
+                top = it.top.toFloat() - topPaddingPx,
+                right = it.right.toFloat(),
+                bottom = it.bottom.toFloat() + bottomPaddingPx
+            )
+        }
+    }
+
+    Layout(
+        modifier = modifier,
+        measurePolicy = foldAwareColumnMeasurePolicy(
+            verticalArrangement = Arrangement.Top,
+            horizontalAlignment = horizontalAlignment,
+            foldBoundsPx = foldBoundsPx
+        ),
+        content = { FoldAwareColumnScopeInstance.content() }
+    )
+}
+
+/**
+ * FoldAwareColumn version of [rowColumnMeasurePolicy] that uses [FoldAwareColumnMeasurementHelper.foldAwarePlaceHelper]
+ * method instead of [RowColumnMeasurementHelper.placeHelper]
+ *
+ * TODO: change from internal to private once metalava issue is solved https://partnerissuetracker.corp.google.com/issues/271539608
```

Comment:
```suggestion
 * TODO: change from internal to private once metalava issue is solved https://issuetracker.google.com/issues/271539608
```

Small nit here to use the publicly visible issue URL.